src/Libraries/Microsoft.Extensions.AI.Ollama/OllamaChatClient.cs

```diff
@@ -356,20 +356,22 @@ private IEnumerable<OllamaChatRequestMessage> ToOllamaChatRequestMessages(ChatMe
                     break;
 
                 case FunctionCallContent fcc:
+                    JsonSerializerOptions serializerOptions = ToolCallJsonSerializerOptions ?? JsonContext.Default.Options;
                     yield return new OllamaChatRequestMessage
                     {
                         Role = "assistant",
                         Content = JsonSerializer.Serialize(new OllamaFunctionCallContent
                         {
                             CallId = fcc.CallId,
                             Name = fcc.Name,
-                            Arguments = FunctionCallHelpers.FormatFunctionParametersAsJsonElement(fcc.Arguments, ToolCallJsonSerializerOptions),
+                            Arguments = JsonSerializer.SerializeToElement(fcc.Arguments, serializerOptions.GetTypeInfo(typeof(IDictionary<string, object?>))),
                         }, JsonContext.Default.OllamaFunctionCallContent)
                     };
                     break;
 
                 case FunctionResultContent frc:
-                    JsonElement jsonResult = FunctionCallHelpers.FormatFunctionResultAsJsonElement(frc.Result, ToolCallJsonSerializerOptions);
+                    JsonSerializerOptions serializerOptions1 = ToolCallJsonSerializerOptions ?? JsonContext.Default.Options;
```

Comment:
You could surround the case bodies where variables are being defined with braces, and then you wouldn't need to use names like serializerOptions1.

Reply:
Can do. I figured I could get away with this to minimize the diff.